src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Sql/Sprocs/UpsertResource_6.sql

```diff
@@ -145,7 +148,14 @@ IF (@previousResourceSurrogateId IS NULL)
     END
 ELSE
     BEGIN
-	-- There is a previous version
+        -- There is a previous version
+        IF (@requireETagOnUpdate = 1
+            AND @etag IS NULL)
+            BEGIN
+                -- This is a versioned update and no version was specified
+                THROW 50412, 'Precondition failed', 1;
+            END
```

Comment:
I don't think this is right place to do.
Imagine doing all the heavy lifting of parsing resource, extracting search values,  filling up tables, sending them to SQL server.

Just to fail on this simple condition.

If we need to fail, fail as soon as possible, as soon as we can check this condition and have all available pieces.